File: typescript/src/model/currencyAmount.ts
import {Check} from '../util/check'

export class CurrencyAmount {

    constructor(readonly currency: string, readonly amount: number) {
        Check.required("currency", currency, Check.format(/[A-Z]{3}/, 'must be a currency code'));
        Check.required("amount", amount);
    }

    public toString(): string {
        return `${this.currency} ${this.amount.toFixed(2)}`;
    }
}

File: python/models/currencyamount.py
from util.check import required, is_number, minimum, is_currency


class CurrencyAmount:
    def __init__(self, currency: str, amount: float):
        self.currency = required('currency', currency, is_currency())
        self.amount = required('amount', amount, is_number(), minimum(0))

    def __str__(self):
        return f'{self.currency} {self.amount:.2f}'
